Src/HALAL/Services/Communication/Ethernet/Ethernet.cpp

```diff
@@ -0,0 +1,97 @@
+/*
+ * Ethernet.cpp
+ *
+ *  Created on: Nov 23, 2022
+ *      Author: stefa
+ */
+
+#include "Communication/Ethernet/Ethernet.hpp"
+
+extern uint32_t EthernetLinkTimer;
+extern struct netif gnetif;
+extern ip4_addr_t ipaddr,netmask,gw;
+
+bool Ethernet::is_ready = false;
+bool Ethernet::is_running = false;
+
+void Ethernet::mpu_start(){
+	  MPU_Region_InitTypeDef MPU_InitStruct = {0};
+
+	  HAL_MPU_Disable();
+
+	  MPU_InitStruct.Enable = MPU_REGION_ENABLE;
+	  MPU_InitStruct.Number = MPU_REGION_NUMBER0;
+	  MPU_InitStruct.BaseAddress = ETHERNET_POOLS_BASE_ADDRESS;
+	  MPU_InitStruct.Size = MPU_REGION_SIZE_32KB;
+	  MPU_InitStruct.SubRegionDisable = 0x0;
+	  MPU_InitStruct.TypeExtField = MPU_TEX_LEVEL1;
+	  MPU_InitStruct.AccessPermission = MPU_REGION_FULL_ACCESS;
+	  MPU_InitStruct.DisableExec = MPU_INSTRUCTION_ACCESS_DISABLE;
+	  MPU_InitStruct.IsShareable = MPU_ACCESS_NOT_SHAREABLE;
+	  MPU_InitStruct.IsCacheable = MPU_ACCESS_NOT_CACHEABLE;
+	  MPU_InitStruct.IsBufferable = MPU_ACCESS_NOT_BUFFERABLE;
+
+	  HAL_MPU_ConfigRegion(&MPU_InitStruct);
+
+	  MPU_InitStruct.Number = MPU_REGION_NUMBER1;
+	  MPU_InitStruct.BaseAddress = ETHERNET_POOLS_BASE_ADDRESS;
+	  MPU_InitStruct.Size = MPU_REGION_SIZE_256B;
+	  MPU_InitStruct.IsShareable = MPU_ACCESS_SHAREABLE;
+
+	  HAL_MPU_ConfigRegion(&MPU_InitStruct);
+
+	  HAL_MPU_Enable(MPU_PRIVILEGED_DEFAULT);
+}
+
+void Ethernet::start(string local_ip, string subnet_mask, string gateway){
+	start(IPV4(local_ip), IPV4(subnet_mask), IPV4(gateway));
+}
+
+void Ethernet::start(IPV4 local_ip, IPV4 subnet_mask, IPV4 gateway){
+	if(!is_running && is_ready){
+		ipaddr = local_ip.ip_address;
+		netmask = subnet_mask.ip_address;
+		gw = gateway.ip_address;
+		MX_LWIP_Init();
+		is_running = true;
+	}else{
+		//TODO: Error Handler
+	}
+}
+
+void Ethernet::inscribe(){
+	if(!is_ready){
```

Comment:
```suggestion
	if(not is_ready){
```

Reply:
no marco el resto, ctrl+F, buscar !, replace not + espacio, miralas una por una no vayas a cambiar un !=

Reply:
no lo voy a hacer eso me da mucha pereza no le veo el sentido entramos en lo que dije de que perdemos el tiempo con cambios estúpidos, entiendo que haya cosas que hagan el código mas inteligible pero estar perdiendo una hora borrando espacios y cambiando operadores no lo voy a hacer he hecho algunas cosas que si que tenian sentido pero no voy a cambiar todos los operadores

Reply:
Abres code, la lupa, te va a buscar y reemplazar para todos los archivos, pones regex, 
![^=] y reemplazas por not espacio, se cambian todos de golpe y no hay error